compositor_pipeline/src/pipeline/whip_whep/validate_bearer_token.rs

```diff
@@ -0,0 +1,42 @@
+use std::time::Duration;
+
+use axum::http::HeaderValue;
+use rand::{rngs::StdRng, Rng, SeedableRng};
+use tokio::time::sleep;
+use tracing::warn;
+
+use crate::pipeline::whip_whep::error::WhipServerError;
+
+pub async fn validate_token(
+    expected_token: Option<String>,
+    auth_header_value: Option<&HeaderValue>,
+) -> Result<(), WhipServerError> {
+    match (expected_token, auth_header_value) {
+        (Some(bearer_token), Some(auth_str)) => {
+            let auth_str = auth_str.to_str().map_err(|_| {
+                WhipServerError::Unauthorized("Invalid UTF-8 in header".to_string())
+            })?;
+
+            if let Some(token_from_header) = auth_str.strip_prefix("Bearer ") {
+                if token_from_header == bearer_token {
+                    Ok(())
+                } else {
+                    let mut rng = StdRng::from_entropy();
+                    let millis = rng.gen_range(50..1000);
+                    sleep(Duration::from_millis(millis)).await;
+                    warn!("Invalid or mismatched token provided");
```

Comment:
Logging warnings/errors is needed when we don't have a way to return that information to the user. In this case this Result will be send in the response,

And if we want to additionally log errors that are sent as resposnses, then why would this error be more important than any other that might be sent in response to this request. We should either log all error responses or none of them